jobsdb/distinct_values_cache.go

```diff
@@ -0,0 +1,115 @@
+package jobsdb
+
+import (
+	"sync"
+
+	"github.com/samber/lo"
+
+	kitsync "github.com/rudderlabs/rudder-go-kit/sync"
+)
+
+func NewDistinctValuesCache() *distinctValuesCache {
+	return &distinctValuesCache{
+		cache: make(map[string]map[string][]string),
```

Comment:
Can we use `sync.Map` instead? It might be a good fit than having multiple locks?

Reply:
a `sync.Map` won't really help loading things once though